File: CPP/Contains_Duplicate.cpp
#include <iostream>
#include <vector>
#include <unordered_set>
#include <unordered_map>
#include <algorithm>


using namespace std;


bool containsDuplicate(vector<int>& nums){
    unordered_set<int> s;
    for(int i=0; i< nums.size(); i++){
        if(s.find(nums[i]) != s.end()){
            return true;
        }
        s.insert(nums[i]);

    }
    return false;
}



//using hash set
bool containsDuplicate(vector<int>& nums) {
    unordered_set<int> s;
    for(int i=0; i< nums.size(); i++){
        if(s.find(nums[i]) != s.end()){
            return true;
        }
        s.insert(nums[i]);

    }
    return false;
}

File: RUST/Contains_Duplicate.rs



pub fn contains_duplicate(nums: Vec<i32>) -> bool {
    
    let mut nums = nums;
    
    nums.sort();
    
    for i in 0..nums.len() - 1 {
    
        if nums[i] == nums[i + 1] {
    
            return true;
    
        }
    
    }
    
    false
    
}


//O(n) time, O(1) space using a hashset
pub fn contains_duplicate(nums: Vec<i32>) -> bool {
    
    let mut set = HashSet::new();
    
    for i in 0..nums.len() {
    
        if set.contains(&nums[i]) {
    
            return true;
    
        }
    
        set.insert(nums[i]);
    
    }
    
    false
    
}